libraries/AP_Math/control.cpp

```diff
@@ -407,3 +407,15 @@ float kinematic_limit(Vector3f direction, float max_xy, float max_z_pos, float m
     }
     return fabsf(max_z_neg/direction.z);
 }
+
+// calculate the expo function on the normalised input
+// input must be in the range of -1 to 1
+// expo should be less than 1.0 but limited to be less than 0.95
+float input_expo(float input, float expo)
+{
+    if (expo < 0.95) {
+        return (1 - expo) * input / (1 - expo * fabsf(input));
```

Comment:
don't we risk divide by zero here if input is 0?

Reply:
yeah, my mistake!

Reply:
I need to add a constrain to ensure the input can't be greater than 1